### PR TITLE
feat: support new 'archived' filter options

### PR DESCRIPTION
**Pull request by cjbell**
This PR adds support for `@knocklabs/client 0.7.0` which adds support for the new `archived` filter.

Additionally it:

- Moves `source` and `tenant` under `defaultFeedOptions` (and expands support to pass `archived` here)
- Add support in `NotificationFeed` for `initialFilterStatus`
- Exposes our (now tested) hooks `useNotifications` and `useAuthenticatedKnockClient`

[Internal ticket](https://linear.app/knock/issue/KNO-1660/[feeds]-support-more-archived-query-options)

### CODE REVIEW

**Review by meryldakin — APPROVED** [review]
didn't test locally but this is looking good to me!

File: src/hooks/useNotifications.ts [hunk 1]
```diff
@@ -20,4 +20,4 @@ function useFeedClient(
   }, [knock, feedId, options.source, options.tenant]);
 }
 
-export default useFeedClient;
+export default useNotifications;
```
Comment (brentjanderson) [inline src/hooks/useNotifications.ts:23]: Is this a breaking change? And does it matter since we're still in major version 0?

**Review by brentjanderson — APPROVED** [review]
LGTM with one question

File: src/hooks/useNotifications.ts (hunk 1 above)
Comment (cjbell) [reply src/hooks/useNotifications.ts:23]: we never exported this hook externally before so the change isn't breaking per-say, but as you said we're not even on a major version yet

**Pull request merged**